models/deepseek_v4_flash_dspark/decode_csa.py

```
@@ -536,6 +556,455 @@ def decode_csa_tp1_test(
+        weights_proj, step_cos_il, step_sin_signed, hadamard_idx,
+        idx_kv_unused, inner_compress_state, inner_compress_state_block_table,
+        inner_wkv, inner_wgate, inner_ape, inner_norm_w,
+        idx_kv_cache, idx_kv_scale, idx_block_table,
+        idx_score_unused, idx_topk_full,
+        position_ids, idx_slot_mapping, inner_state_slot_mapping,
+        kv_seq_lens, 0, late_dep,
+    )
+
+    position_ids_t1 = pl.reshape(position_ids, [t_dim, 1])
+    attention_grouped = pl.create_tensor([O_GROUPS * LOCAL_T_PAD, O_GROUP_IN], dtype=pl.BF16)
+    attention_grouped, heads_tid = sparse_attn_csa(
```

Ruff (0.16.1)

[warning] Unpacked variable `heads_tid` is never used

Prefix it with an underscore or any other dummy variable pattern

(RUF059)